core/include/kernel_defines.h

```diff
@@ -138,6 +138,61 @@
  */
 #define BUILD_BUG_ON(condition) ((void)sizeof(char[1 - 2 * !!(condition)]))
 
+/**
+ * @def         IS_ACTIVE(option)
+ * @brief       Allows to verify a macro definition outside the preprocessor.
+ *
+ * @details     This macro is based on Linux's clever 'IS_BUILTIN'
+ *              (https://github.com/torvalds/linux/blob/master/include/linux/kconfig.h).
+ *              It takes a @p macro value that may be defined to 1 or not even
+ *              defined (e.g. MODULE_NETIF) and then expands it to an expression
+ *              that can be used in C code, either 1 or 0.
+ *
+ *              The advantage of using this is that the compiler sees all the
+ *              code, so checks can be performed, sections that would not be
+ *              executed are removed during optimization. For example:
+ * ```
+ * if (IS_ACTIVE(MODULE_NETIF)) {
+ *      do_something();
+ * }
+ * ```
+ * @param[in]   macro   Macro to evaluate
+ * @returns     1 if the macro is defined to 1
+ * @returns     0 if the macro is not defined, of if it is defined to something
+ *              else than 1.
+ *
+ * @note        Although this may seem to work similarly to the preprocessor's
+ *              'defined', it is not entirely equal. If the given macro has
+ *              been defined with no value, this will expand to 0. Also note
+ *              that this is intended to be used with 'boolean' macros that act
+ *              as switches, and usually will be defined as 1 or not defined.
+ */
+#define IS_ACTIVE(macro) __is_active(macro)
```

Comment:
A bit of bike-shedding: I do not like the name. With `USEMODULE +=` used to enable module, I would like something like `IS_MODULE_USED()`, or `IS_MOD_USED()`, or `MODULE_USED()`, or ...

Reply:
Could be used with things that are not just `MODULE_` macros, maybe a broader name?

Reply:
That is true. Also the term "module" is given as argument. `IS_MODULE_USED(MODULE_SHT)` would read a bit strange...